Sources/MobileSdk/MDocHolderBLECentral.swift

```diff
@@ -17,20 +19,45 @@ enum DataError: Error {
     case unknownDataTransferPrefix(byte: UInt8)
 }
 
+/// The MDoc holder as a BLE central.
 class MDocHolderBLECentral: NSObject {
+    enum MachineState {
+        case initial, hardwareOn, fatalError, complete, halted
+        case awaitPeripheralDiscovery, peripheralDiscovered, checkPeripheral
+        case awaitRequest, requestReceived, sendingResponse
+        case l2capAwaitRequest, l2capRequestReceived, l2capSendingResponse
+    }
+
     var centralManager: CBCentralManager!
     var serviceUuid: CBUUID
     var callback: MDocBLEDelegate
     var peripheral: CBPeripheral?
+
     var writeCharacteristic: CBCharacteristic?
     var readCharacteristic: CBCharacteristic?
     var stateCharacteristic: CBCharacteristic?
+    var l2capCharacteristic:  CBCharacteristic?
+
     var maximumCharacteristicSize: Int?
     var writingQueueTotalChunks = 0
     var writingQueueChunkIndex = 0
     var writingQueue: IndexingIterator<ChunksOfCountCollection<Data>>?
 
     var incomingMessageBuffer = Data()
+    var outgoingMessageBuffer = Data()
+
+    private var channelPSM: UInt16? = nil
+    private var activeStream: MDocHolderBLECentralConnection?
+
+    /// If this is `false`, we decline to connect to L2CAP even if it is offered.
+    var useL2CAP = true
```

Comment:
This should be configurable all the way through `BLESessionManager `

Reply:
Like the one in the reader, `useL2CAP` was really for testing; I included them so I could ensure that if I manually shut off support either in the reader or the holder, it would fall back on the old flow without problem.  I made them `var` and not `private` on the assumption that we might want this for automated testing as well, but I'm not sure we want them to be part of the API, per se.

Reply:
I reckon we do want it as part of the API because let's say we favour L2CAP as it is faster -- it's very possible that some readers end up being incompatible with our implementation so it would be good for holders using our library to be able to disable it entirely as a stop-gap solution

Reply:
The decision of whether to use the L2CAP flow or the old flow is entirely in the holder's hands (or really, in the central's hands).  Peripheral offers both the L2CAP and State characteristics, and Central picks one to talk to; the rest of the flow is chosen by which characteristic Central initially chose to engage. A non-L2CAP peripheral simply doesn't offer the L2CAP characteristic, so Central has no choice and the old flow is used.  An L2CAP supporting peripheral offers the L2CAP characteristic alongside the others, but Central can choose to ignore the L2CAP characteristic entirely and engage State, at which point the old flow is used.
Incompatibility is a real possibility, though, as you say.

Reply:
As it currently stands, `BLESessionManager` can reach in and set `useL2CAP` manually, so we could just have a call in the session manager to do it. Or do we want to add a `L2CAP: Boolean` argument to the reader/holder init?

Reply:
I would add it as part of `presentMdocBLE` as a parameter with a default value to use L2CAP if available

Reply:
Done, with a similar default arg for the reader.